src/test/java/software/amazon/encryption/s3/S3EncryptionClientInstructionFileTest.java

```diff
@@ -0,0 +1,458 @@
+
+        S3Client wrappedClient = S3Client.create();
+        S3Client s3Client = S3EncryptionClient.builder()
+                .instructionFileConfig(InstructionFileConfig.builder()
+                        .instructionFileClient(wrappedClient)
+                        .enableInstructionFilePutObject(true)
+                        .build())
+                .kmsKeyId(KMS_KEY_ID)
+                .build();
+
+        Map<String, String> encryptionContext = new HashMap<>();
+        encryptionContext.put("test-key", "test-value");
+
+
+        s3Client.putObject(builder -> builder
+                .bucket(BUCKET)
+                .overrideConfiguration(withAdditionalConfiguration(encryptionContext))
+                .key(object_key), RequestBody.fromInputStream(inputStream, fileSizeLimit));
+
+        S3Client defaultClient = S3Client.create();
+        ResponseBytes<GetObjectResponse> directInstGetResponse = defaultClient.getObjectAsBytes(builder -> builder
+                .bucket(BUCKET)
+                .key(object_key + ".instruction")
+                .build());
+        assertTrue(directInstGetResponse.response().metadata().containsKey("x-amz-crypto-instr-file"));
+
+        ResponseInputStream<GetObjectResponse> getResponse = s3Client.getObject(builder -> builder
+                .bucket(BUCKET)
+                .overrideConfiguration(withAdditionalConfiguration(encryptionContext))
+                .key(object_key));
+
+        assertTrue(IOUtils.contentEquals(objectStreamForResult, getResponse));
+
+        deleteObject(BUCKET, object_key, s3Client);
+        s3Client.close();
+
+    }
+
+    @Test
+    public void testLowLevelMultipartPutWithInstructionFile() throws NoSuchAlgorithmException, IOException {
+        final String object_key = appendTestSuffix("test-low-level-multipart-put-instruction-file");
+
+        final long fileSizeLimit = 1024 * 1024 * 50;
+        final int PART_SIZE = 10 * 1024 * 1024;
+        final InputStream inputStream = new BoundedInputStream(fileSizeLimit);
+        final InputStream objectStreamForResult = new BoundedInputStream(fileSizeLimit);
+
+        KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA");
+        keyPairGen.initialize(2048);
+        KeyPair rsaKey = keyPairGen.generateKeyPair();
+
+        S3Client wrappedClient = S3Client.create();
+
+        S3Client v3Client = S3EncryptionClient.builder()
+                .rsaKeyPair(rsaKey)
+                .instructionFileConfig(InstructionFileConfig.builder()
+                        .instructionFileClient(wrappedClient)
+                        .enableInstructionFilePutObject(true)
+                        .build())
+                .enableDelayedAuthenticationMode(true)
+                .build();
+
+
+        CreateMultipartUploadResponse initiateResult = v3Client.createMultipartUpload(builder ->
+                builder.bucket(BUCKET).key(object_key));
```

Comment:
Let's also set `StorageClass` here and check for it in the `directInstGetResponse` later on.